src/Core/Math/LinearAlgebra.hpp

```diff
@@ -99,26 +112,27 @@ inline Scalar getNormAndNormalize( Vector_& v );
 template <typename Vector_>
 inline Scalar getNormAndNormalizeSafe( Vector_& v );
 
-} // namespace Vector
+/// Transform a ray, direction is only transformed by linear part of the
+/// transformation, while origin is fully transformed
+/// corresponds to t*r
+/// \param t : transform matrix
+/// \param r : ray to transform
+/// \return transoformed ray, origine is translated while direction is only
+/// linarly transformed.
+template <typename Scalar>
+inline Eigen::ParametrizedLine<Scalar, 3>
```

Comment:
Why not using `Ray` ?

Reply:
Because you can't template on Scalar type when using Ray.
this function is templated so it is meaningfull for Ray, Rayf, Rayd.